generator/types.go

```diff
@@ -62,8 +66,22 @@ func (p Param) Pass() string {
 }
 
 // NewMethod returns pointer to Signature struct or error
-func NewMethod(name string, f *ast.FuncType, printer typePrinter) (*Method, error) {
+func NewMethod(name string, f *ast.FuncType, printer typePrinter, fi *ast.Field) (*Method, error) {
```

Comment:
I think we can replace name, f and fi with just field *ast.Field as a first argument and get everything we need inside the constructor.